### DISCUSSION TITLE
Possibility to obtain csv file preserving the complete history of the reviewing process

### DISCUSSION THREAD

**Discussion by BartJanBoverhof** [asreview/asreview#1089] [Ideas]
Hello!

Is it possible to easily obtain the complete history of a review conducted in `asreview lab` mode? 
In order for doing some data visualisations, I am looking for a way to export my finished project into a `.csv` file in which the complete history of my reviewing process is preserved. I would like this in order to be able to freely work on visualisation of the reviewing process myself. Specifically I am looking for information on:

- Record flagged as relevant (yes/no)
- For each record, the exact order in which it was shown over the course of the reviewing process. 
- Whether the record was shown at all (yes/no)

I am aware that some visualisation can also be done with the [asreview visualisation extension](https://github.com/asreview/asreview-visualization), however, I would like to have some more freedom in doing the plotting from scratch.

_(In case this is not easily possible I'd like to frame this post as feature request)_

**Comment by J535D165** [comment]
Thanks, would be a nice feature to have. This is way more straightforward to implement in version v1 (checkout master branch for release candidate). In v1, the state file contains a sqlite3 table named `results` containing all label decisions in the same order as the labeling by the user. A simple join with the dataset would result in the requested feature (including more info like model used and notes). 

It would be interesting to see/document a code example. We are welcoming contributions on this!
